### PR TITLE
Schema v0.7

### PR DESCRIPTION
**Pull request by makeworld-the-better-one**
This includes work from #30, #28, and, #29, and fully passes CI.

Additionally:
- Added `routerModel`
- Made `status` mandatory

Is there anything else that v0.7 should include?

### CODE REVIEW

File: nodeList.json [hunk 1]
```diff
@@ -1,13 +1,16 @@
 {
   "nodeList": [
     {
+      "routerModel": "edgerouter-x-sfp",
```
Comment (darkdrgn2k) [inline nodeList.json:4]: Should this just be collapsed to "model" for both "routerModel" and "antennaModel"

File: schema/v0.7/schema.json [hunk 2]
```diff
@@ -0,0 +1,243 @@
...
+                    "antennaDistance": {
+                        "$id": "#root/nodeList/items/antennaDistance", 
+                        "title": "Antenna Distance", 
+                        "type": "integer",
+                        "examples": [
+                            2500
+                        ],
+                        "default": 0
+                    },
+                    "antennaModel": {
```
Comment (darkdrgn2k) [inline schema/v0.7/schema.json:68]: collapse to "model"

File: schema/v0.7/schema.json [hunk 3]
```diff
@@ -0,0 +1,243 @@
...
+                        "$id": "#root/nodeList/items/type", 
+                        "title": "Type", 
+                        "type": "string",
+                        "default": "",
+                        "examples": [
+                            "antenna"
+                        ],
+                        "pattern": "^.*$"
+                    },
+                    "routerModel": {
```
Comment (darkdrgn2k) [inline schema/v0.7/schema.json:228]: Collapse to "model"

**Comment by darkdrgn2k** [comment]
> Is there anything else that v0.7 should include?

This is rapidly evolving, id rather not be versioning them as lost of this is adhoc.

Adding "ssid-client" for P2P links that will be used for thorncliffe

Also i dont know if people's nodes should be in this file or in some other file.

File: nodeList.json (hunk 1 above)
Comment (darkdrgn2k) [inline nodeList.json:4]:
```suggestion
      "model": "er-x-sfp",
```

Collapse  models into one field
ex-x-sfp is the model number on the box

File: schema/v0.7/schema.json (hunk 2 above)
Comment (darkdrgn2k) [inline schema/v0.7/schema.json:68]:
```suggestion
                    "model": {
```

File: schema/v0.7/schema.json [hunk 4]
```diff
@@ -0,0 +1,243 @@
...
+                        "$id": "#root/nodeList/items/antennaDistance", 
+                        "title": "Antenna Distance", 
+                        "type": "integer",
+                        "examples": [
+                            2500
+                        ],
+                        "default": 0
+                    },
+                    "antennaModel": {
+                        "$id": "#root/nodeList/items/antennaModel",
```
Comment (darkdrgn2k) [inline schema/v0.7/schema.json:69]:
```suggestion
                        "$id": "#root/nodeList/items/model", 
```

File: schema/v0.7/schema.json [hunk 5]
```diff
@@ -0,0 +1,243 @@
...
+                        "title": "Antenna Distance", 
+                        "type": "integer",
+                        "examples": [
+                            2500
+                        ],
+                        "default": 0
+                    },
+                    "antennaModel": {
+                        "$id": "#root/nodeList/items/antennaModel", 
+                        "title": "Antenna Model",
```
Comment (darkdrgn2k) [inline schema/v0.7/schema.json:70]:
```suggestion
                        "title": "Model Number", 
```

File: schema/v0.7/schema.json [hunk 6]
```diff
@@ -0,0 +1,243 @@
...
+                        ],
+                        "default": 0
+                    },
+                    "antennaModel": {
+                        "$id": "#root/nodeList/items/antennaModel", 
+                        "title": "Antenna Model", 
+                        "type": "string",
+                        "default": "",
+                        "examples": [
+                            "lap-120"
```
Comment (darkdrgn2k) [inline schema/v0.7/schema.json:74]:
```suggestion
                            "lap-120",
                            "er-x-sfp"
```

File: schema/v0.7/schema.json [hunk 7]
```diff
@@ -0,0 +1,243 @@
...
+                        ],
+                        "default": 0.0
+                    },
+                    "mode": {
+                        "$id": "#root/nodeList/items/mode", 
+                        "title": "Mode", 
+                        "type": "string",
+                        "default": "",
+                        "examples": [
+                            "ap"
```
Comment (darkdrgn2k) [inline schema/v0.7/schema.json:172]:
```suggestion
                            "ap",
                            "station",
                            "adhoc",
                            "802.11s"
```

File: schema/v0.7/schema.json [hunk 8]
```diff
@@ -0,0 +1,243 @@
...
+                    "routerModel": {
+                        "$id": "#root/nodeList/items/routerModel", 
+                        "title": "Router Model", 
+                        "type": "string",
+                        "default": "",
+                        "examples": [
+                            "edgerouter-x-sfp"
+                        ],
+                        "pattern": "^.*$"
+                    }
```
Comment (darkdrgn2k) [inline schema/v0.7/schema.json:237]:
```suggestion
```

File: schema/v0.7/schema.json (hunk 2 above)
Comment (darkdrgn2k) [reply schema/v0.7/schema.json:68]: This section probably needs to move down to be alphabetical

File: schema/v0.7/schema.json (hunk 5 above)
Comment (makeworld-the-better-one) [reply schema/v0.7/schema.json]: I changed this to `Hardware Model`, if that's good.

**Comment by makeworld-the-better-one** [comment]
> This is rapidly evolving, id rather not be versioning them as lost of this is adhoc.
 
I agree. I've moved the `schema.json` file out of the `0.7` dir, see 0e14a1a.

> ~Adding "ssid-client" for P2P links that will be used for thorncliffe~
> Normalised into single ssid

I will leave the single SSID key then. I remeber you mentioning MAC keys on the chat. I don't have the data for those, please add them if you think they'd be good.

File: README.md [hunk 9]
```diff
@@ -3,12 +3,25 @@
 [![Build Status](https://travis-ci.org/tomeshnet/node-list.svg?branch=master)](https://travis-ci.org/tomeshnet/node-list)
 
 This repository contains the list of nodes that are part of the Toronto Community Network.
-Each node is recorded with geolocation and technical information as according to [schema/v0.6/schema.json](schema/v0.6/schema.json).
-The list is organized as a JSON file at [nodeList.json](nodeList.json).
+Each node is recorded with geolocation and technical information as according to [schema.json][schema-json].
+The list of nodes is organized as a JSON file at [nodeList.json](nodeList.json).
+
+## Images
```
Comment (benhylau) [inline README.md:9]: Better here

File: .travis.yml [hunk 10]
```diff
@@ -3,6 +3,7 @@ sudo: false
 rvm:
   - 2.4.6
 
+before_script: export SCHEMA_VERSION=0.7
```
Comment (benhylau) [inline .travis.yml:6]: Let's define the specific version we are validating for here and not mess with the file structure. @makeworld-the-better-one in your proposal the current version is no in the file path, we don't want that.

File: README.md [hunk 11]
```diff
@@ -3,12 +3,25 @@
...
 
-In this repository, [Travis CI](.travis.yml) is configured to validate [nodeList.json](nodeList.json) against [schema/v0.6/schema.json](schema/v0.6/schema.json) using [Ruby JSON Schema Validator](https://github.com/ruby-json-schema/json-schema).
+In this repository, [Travis CI](.travis.yml) is configured to validate [nodeList.json](nodeList.json) against [schema.json][schema-json] using [Ruby JSON Schema Validator](https://github.com/ruby-json-schema/json-schema).
 The validation tool currently supports up to [JSON Schema Draft 4](https://tools.ietf.org/html/draft-zyp-json-schema-04), so schema files must conform to the Draft 4 standard.
 
-You can use [JSON Schema Lint](https://jsonschemalint.com) to validate manually.
\ No newline at end of file
+You can use [JSON Schema Lint](https://jsonschemalint.com) to validate manually.
+
+[schema-json]: schema/v0.7/schema.json
```
Comment (benhylau) [inline README.md:27]: Here needs to stay in sync with the travis.

File: nodeList.json [hunk 12]
```diff
@@ -2,51 +2,65 @@
   "nodeList": [
     {
       "altitude": 170,
-      "dateAdded": "2020-07-23T18:00:00.000Z",
+      "dateAdded": "2020-07-23",
```
Comment (benhylau) [inline nodeList.json:5]: @YurkoWasHere we should change this to date only. The schema now validates it as such.

File: schema/v0.7/schema.json [hunk 13]
```diff
@@ -0,0 +1,238 @@
...
+                        "default": "",
+                        "examples": [
+                            "ap",
+                            "station",
+                            "adhoc",
+                            "802.11s"
+                        ],
+                        "pattern": "^.*$"
+                    },
+                    "model": {
```
Comment (benhylau) [inline schema/v0.7/schema.json:169]: `antennaModel` is now gone. It's just model.

File: schema/v0.7/schema.json [hunk 14]
```diff
@@ -0,0 +1,238 @@
...
+                        "$id": "#root/nodeList/items/status", 
+                        "title": "Status", 
+                        "type": "string",
+                        "default": "active",
+                        "examples": [
+                            "active",
+                            "inactive", 
+                            "proposed"
+                        ],
+                        "pattern": "^active|inactive|proposed$"
```
Comment (benhylau) [inline schema/v0.7/schema.json:220]: A lot of the values are strictly validated now. This allows UI designer to handle only a subset of the infinite possibilities called `string`.

File: schema/v0.7/schema.json [hunk 15]
```diff
@@ -0,0 +1,238 @@
...
+                    },
+                    "ipv4": {
+                        "$id": "#root/nodeList/items/ipv4", 
+                        "title": "IPv4", 
+                        "type": "string",
+                        "default": "",
+                        "examples": [
+                            "100.64.10.2"
+                        ],
+                        "format": "ipv4"
```
Comment (benhylau) [inline schema/v0.7/schema.json:126]: v4 and v6 are strictly validated now.

File: schema/v0.7/schema.json [hunk 16]
```diff
@@ -0,0 +1,238 @@
...
+                    },
+                    "dateAdded": {
+                        "$id": "#root/nodeList/items/dateAdded", 
+                        "title": "Date Added", 
+                        "type": "string",
+                        "default": "",
+                        "examples": [
+                            "2020-07-23"
+                        ],
+                        "format": "date"
```
Comment (benhylau) [inline schema/v0.7/schema.json:96]: I opted for `date` not `date-time`.

File: spec/schema.rb [hunk 17]
```diff
@@ -1,11 +1,11 @@
 require 'json-schema'
 
-schema_version = 0.6
+schema_version = ENV['SCHEMA_VERSION']
```
Comment (benhylau) [inline spec/schema.rb:3]: Now this is set via env var.

**Comment by makeworld-the-better-one** [comment]
Waiting on @YurkoWasHere to add MAC keys if he would like to. After that I feel good to merge.

**Comment by benhylau** [comment]
@YurkoWasHere if you like to add mac just add to nodeList and I will take care of schema.

Pls also review the notes I left in #5 #25 if there is anything you want in v0.7.

File: nodeList.json (hunk 12 above)
Comment (darkdrgn2k) [reply nodeList.json:5]: sure, time is annoying to calculate anywa

**Comment by darkdrgn2k** [comment]
#5 - No dynamic data in nodelist Static only
#25 - dont know if this is the right context anymore.  Perhaps we need to build a SERVICES json file.

**Comment by darkdrgn2k** [comment]
MACS:

Don't know if this should be stored separately or reported dynamically or maybe both?

**Comment by makeworld-the-better-one** [comment]
My understanding is that MACs are mostly static, at least in our case. If I'm not missing anything, I don't see an issue with having them in this file. Or v0.8 if integrating that will take some time.

**Comment by benhylau** [comment]
In my opinion MAC is tricky bc of privacy concerns as they are generally static. However, devices with softmacs can change their MACs which make it a dynamic property. Let's leave this for later.

**Pull request merged**